File: examples/flutter/lib/products_screen.dart
import 'package:flutter/material.dart';

/// Products list — navigated to from HomeScreen.
///
/// Does not call [UXRate.setScreen] to keep the demo focused on the main tabs.
/// In a real app you would add a setScreen call here if you want surveys on this screen.
class ProductsScreen extends StatelessWidget {
  const ProductsScreen({super.key});

  static const _products = [
    _Product('MacBook Pro', Icons.laptop_mac, '\$2,499'),
    _Product('iPhone', Icons.phone_iphone, '\$999'),
    _Product('AirPods Pro', Icons.headphones, '\$249'),
    _Product('Apple Watch', Icons.watch, '\$399'),
    _Product('iPad Pro', Icons.tablet_mac, '\$1,099'),
  ];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('Products')),
      body: ListView.separated(
        itemCount: _products.length,
        separatorBuilder: (_, __) => const Divider(height: 1),
        itemBuilder: (context, index) {
          final product = _products[index];
          return ListTile(
            leading: Icon(product.icon, color: Theme.of(context).colorScheme.primary),
            title: Text(product.name),
            trailing: Text(
              product.price,
              style: Theme.of(context).textTheme.bodyMedium?.copyWith(
                    fontWeight: FontWeight.w600,
                  ),
            ),
          );
        },
      ),
    );
  }
}

class _Product {
  const _Product(this.name, this.icon, this.price);
  final String name;
  final IconData icon;
  final String price;
}

File: examples/android/app/src/main/java/com/uxrate/demo/ProductsScreen.kt
package com.uxrate.demo

import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.ListItem
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight

/**
 * Products list — navigated to from HomeScreen.
 *
 * Does not call UXRate.setScreen() to keep the demo focused on the main tabs.
 * In a real app you would add a setScreen call here if you want surveys on this screen.
 */
@Composable
fun ProductsScreen() {
    val products = listOf(
        Product("MacBook Pro", "💻", "$2,499"),
        Product("iPhone", "📱", "$999"),
        Product("AirPods Pro", "🎧", "$249"),
        Product("Apple Watch", "⌚", "$399"),
        Product("iPad Pro", "📲", "$1,099"),
    )

    LazyColumn(modifier = Modifier.fillMaxSize()) {
        items(products) { product ->
            ListItem(
                headlineContent = { Text(product.name) },
                leadingContent = { Text(product.emoji, style = MaterialTheme.typography.headlineMedium) },
                trailingContent = {
                    Text(
                        product.price,
                        fontWeight = FontWeight.SemiBold,
                    )
                },
            )
            HorizontalDivider()
        }
    }
}

private data class Product(val name: String, val emoji: String, val price: String)
